assets/js/modules/ads/components/PAXEmbeddedApp.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * Ads Partner Ads Experience (PAX) Embedded App component.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useInterval } from 'react-use';
+
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+import {
+	useCallback,
+	useEffect,
+	useMemo,
+	useRef,
+	useState,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_USER } from '../../../googlesitekit/datastore/user/constants';
+import CTA from '../../../components/notifications/CTA';
+import PreviewBlock from '../../../components/PreviewBlock';
+import { createPaxServices } from '../pax/services';
+const { useRegistry, useSelect } = Data;
+
+export default function PAXEmbeddedApp( {
+	// eslint-disable-next-line no-unused-vars
+	displayMode = 'default',
+	onLaunch,
+} ) {
+	const [ launchGoogleAdsAvailable, setLaunchGoogleAdsAvailable ] = useState(
+		typeof global?.google?.ads?.integration?.integrator?.launchGoogleAds ===
+			'function'
+	);
+
+	const [ hasLaunchedPAXApp, setHasLaunchedPAXApp ] = useState( false );
+	const [ isLoading, setIsLoading ] = useState( true );
+	const [ launchError, setLaunchError ] = useState( undefined );
+
+	const registry = useRegistry();
+
+	const paxServices = createPaxServices( registry );
+
+	const isAdBlockerActive = useSelect( ( select ) =>
+		select( CORE_USER ).isAdBlockerActive()
+	);
+
+	const instanceID = useInstanceId( PAXEmbeddedApp, 'PAXEmbeddedApp' );
+
+	const paxAppRef = useRef();
+
+	const elementID = useMemo( () => {
+		return `googlesitekit-pax-embedded-app-${ instanceID }`;
+	}, [ instanceID ] );
+
+	const paxConfig = useMemo( () => {
+		return {
+			...( global?._googlesitekitPAXConfig || {} ),
+			clientConfig: {
+				contentContainer: `#${ elementID }`,
+			},
+		};
+	}, [ elementID ] );
+
+	const paxServicesWithAuthToken = useMemo( () => {
+		return {
+			authenticationService: {
```

Comment:
This service wasn't required as part of this issue, and requires a bit more thought as it's only called when the token we provide in the config has expired. Providing the same token it already has wouldn't be correct.

We probably need a new issue to fetch this from the API and then this service can be defined with the rest of them.

Reply:
Cool, I'll take this out for now 👍🏻